parsl/executors/radical/executor.py

```diff
@@ -0,0 +1,388 @@
+                task.arguments = ['-c', bash_app]
+
+                # specifying pre_exec is only for executables
+                task.pre_exec = kwargs.get('pre_exec', [])
+
+        elif PYTHON in task_type or not task_type:
+            task.mode = rp.TASK_FUNCTION
+            task.scheduler = 'master.%06d' % (tid % self.n_masters)
+            task.function = PythonTask(func, *args, **kwargs)
+
+        task.stdout = kwargs.get('stdout', '')
+        task.stderr = kwargs.get('stderr', '')
+        task.cpu_threads = kwargs.get('cpu_threads', 1)
+        task.gpu_threads = kwargs.get('gpu_threads', 0)
+        task.cpu_processes = kwargs.get('cpu_processes', 1)
+        task.gpu_processes = kwargs.get('gpu_processes', 0)
+        task.cpu_process_type = kwargs.get('cpu_process_type', '')
+        task.gpu_process_type = kwargs.get('gpu_process_type', '')
+
+        return task
+
+    def _bulk_collector(self):
+
+        bulk = list()
+
+        while True:
+
+            now = time.time()  # time of last submission
+
+            # collect tasks for min bulk time
+            # NOTE: total collect time could actually be max_time + min_time
+            while time.time() - now < self._max_bulk_time:
+
+                try:
+                    task = self._bulk_queue.get(block=True, timeout=self._min_bulk_time)
+                except queue.Empty:
+                    task = None
+
+                if task:
+                    bulk.append(task)
+
+                if len(bulk) >= self._max_bulk_size:
+                    break
+
+            if bulk:
+                logger.debug('submit bulk: %d', len(bulk))
+                self.tmgr.submit_tasks(bulk)
+                bulk = list()
+
+    def submit(self, func, *args, **kwargs):
+        """
+        Submits tasks iteratively (stream mode) or tasks in bulks (bulk mode) to
+        RADICAL task_manager.
+
+        Args:
+            - func (callable) : Callable function
+            - *args (list)    : List of arbitrary positional arguments.
+
+        Kwargs:
+            - **kwargs (dict) : A dictionary of arbitrary keyword args for func.
+        """
+        rp_tid = ru.generate_id('task.%(item_counter)06d', ru.ID_CUSTOM,
+                                ns=self.session.uid)
+        parsl_tid = int(rp_tid.split('task.')[1])
+
+        logger.debug("got {0} from parsl-dfk".format(parsl_tid))
+        task = self.task_translate(parsl_tid, func, args, kwargs)
+
+        # assign task id for rp task
+        task.uid = rp_tid
+
+        # set the future with corresponding id
+        self.future_tasks[rp_tid] = Future()
+
+        if self.bulk_mode:
+            # push task to rp submit thread
+            self._bulk_queue.put(task)
+        else:
+            # submit the task to rp
+            logger.debug("put {0} to rp-TMGR".format(rp_tid))
+            self.tmgr.submit_tasks(task)
+
+        return self.future_tasks[rp_tid]
+
+    def _get_job_ids(self):
```

Comment:
this method was changed in PR #1952 (commit 5eb84c0859929110b06e72d4843d45a9633157a2) 
to this:
 +    def _get_block_and_job_ids(self) -> Tuple[List[str], List[Any]]:

but I think you don't need it at all as this class inherits from `NoStatusHandlingExecutor` which does not do any block management.

Reply:
Agreed and removed, thanks.